text/0000-system-node.md

```diff
@@ -0,0 +1,108 @@
+
+For all the above reasons, **the best approach to using a Node version manager is to avoid having any other Node installations on the system**. Rather than attempting to hide the risks associated with co-installing along with a system Node, we should clearly recommend to users of Notion that if they have an existing system installation of Node, they should uninstall it first.
+
+## Unix and Windows
+
+In Unix OSes, we may want to consider still allowing the ability to opt into overriding a system installation. But since this incurs greater security risk and is generally less reliable, it should not be the recommended path. The installation steps should ask for confirmation and should recommend aborting and choose this as the default action.
+
+In Windows, the system `Path` variable takes precedence over the user `Path` variable---not dissimilar from how Unix environments place `/usr/bin` towards the beginning of `PATH`. But unlike Unix, Windows provides no general mechanism to circumvent this (whereas Unix users can simply prepend new directories to the front of `PATH`). This suggests that the Windows design has a stronger expectation that system installations take precedence over user installations.
+
+Also unlike Unix, Windows developers do not expect environment variables to be configured by terminal bootup scripts. Windows users should be able to expect use cases like executing Node from "Start | Run" or in IDEs to work, which would not work with terminal configuration scripts.
+
+Rather than attempting to circumvent standard practice or provide partial solutions that don't work for all expected contexts, the Windows Notion installer should simply fail if a system Node is detected.
+
+## Detecting system changes
+
+Detecting a system Node at installation time is insufficient since users or system administrators could install another Node installation on the system at any time. So we also need to provide users with the tools to detect these kinds of issues at any time after Notion has already been installed.
+
+# Pedagogy
+[pedagogy]: #pedagogy
+
+The simplest story to explain to users is: **don't install Notion alongside other Node installations**.
+
+With override, we can soften this to a recommendation, and optionally allow overriding.
+
+The installation UI provides one opportunity to explain this; it's also worth recommending on the Notion web site.
+
+The idea that Notion installs the toolchain in the user's own space should not be too difficult to learn, since it's already the dominant model for many toolchain managers.
+
+# Details
+[details]: #details
+
+## Installation
+
+At installation time, the installer should check for:
+
+- Existing `node`, `npm`, `npx`, or `yarn` commands in the path.
```

Comment:
These commands may also be in the path if the user has another node version manager installed (nvm, n, nodebrew, avn, &c.). Maybe a topic for a separate RFC, but should notion recommend against being installed alongside other version managers? Or is there a way that notion can coexist with other version managers?

Reply:
Good catch! This RFC should probably discuss that as well.

My sense is that it's going to be just as problematic to have another version manager active in your `PATH`. I'll chew on this while I commute to try to come up with some example problem cases…